app/forms/item/additionalReference/AdditionalReferenceNumberFormProvider.scala

```diff
@@ -18,21 +18,23 @@ package forms.item.additionalReference
 
 import config.PhaseConfig
 import forms.mappings.Mappings
+import models.Phase
 import models.domain.StringFieldRegex.stringFieldRegexComma
 import play.api.data.Form
 
 import javax.inject.Inject
 
 class AdditionalReferenceNumberFormProvider @Inject() (implicit phaseConfig: PhaseConfig) extends Mappings {
 
-  def apply(prefix: String, otherAdditionalReferenceNumbers: Seq[String]): Form[String] =
+  def apply(prefix: String, otherAdditionalReferenceNumbers: Seq[String], isDocumentInCL234: Option[Boolean], phase: Phase): Form[String] =
     Form(
       "value" -> text(s"$prefix.error.required")
         .verifying(
           forms.StopOnFirstFail[String](
             regexp(stringFieldRegexComma, s"$prefix.error.invalidCharacters"),
             maxLength(phaseConfig.maxAdditionalReferenceNumLength, s"$prefix.error.length"),
-            notInList(otherAdditionalReferenceNumbers, s"$prefix.error.unique")
+            notInList(otherAdditionalReferenceNumbers, s"$prefix.error.unique"),
+            cl234Constraint(isDocumentInCL234.getOrElse(false), phase, s"$prefix.error.cl234Constraint")
```

Comment:
Could we do a `getMandatoryPage` on `AdditionalReferenceInCL234Page` in the `AdditionalReferenceNumberController` so we don't need to pass it in as an option?